Main.py

```diff
@@ -23,7 +23,7 @@
 
 def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = None):
     if not baked_server_options:
-        baked_server_options = get_settings().server_options
+        baked_server_options = get_settings().server_options.as_dict()
```

Comment:
I would suggest asserting that it's actually a dict so we might catch other errors
```suggestion
        baked_server_options = get_settings().server_options.as_dict()
    assert isinstance(baked_server_options, dict)
```